src/Avalonia.Controls/Platform/MacOSProperties.cs

```diff
@@ -10,6 +12,7 @@ public class MacOSProperties
     static MacOSProperties()
     {
         IsTemplateIconProperty.Changed.AddClassHandler<TrayIcon>(TrayIconIsTemplateIconChanged);
+        CanBecomeKeyWindowProperty.Changed.AddClassHandler<Popup>(CanBecomeKeyWindowChanged);
```

Comment:
You can override default value for Window and Popup independently. And handle events for both, making it technically usable with any window. Although, I don't know where it's useful on non-popups.